vms/administrator/views.py

```diff
@@ -1,62 +1,66 @@
 from administrator.forms import ReportForm
-
 from django.contrib.auth.decorators import login_required
 from django.core.exceptions import ObjectDoesNotExist
 from django.core.urlresolvers import reverse
 from django.http import Http404, HttpResponse, HttpResponseRedirect
 from django.shortcuts import render
-
 from shift.services import *
 from event.services import *
 from job.services import *
+from django.views.generic import TemplateView
+from django.views.generic import ListView
+from braces.views import LoginRequiredMixin, AnonymousRequiredMixin
+from django.views.generic.edit import FormView, UpdateView
+from django.shortcuts import render_to_response
+from django.views.generic import View
+from administrator.models import Administrator
 
-@login_required
-def report(request):
 
-    user = request.user
-    admin = None
-    try:
-        admin = user.administrator
-    except ObjectDoesNotExist:
-        pass
-    if not admin:
-        return render(request, 'vms/no_admin_rights.html')
+#Displays the form
+class ShowFormView(LoginRequiredMixin, FormView):
+    model = Administrator
+    form_class = ReportForm
+    template_name = "administrator/report.html"
 
```

Comment:
@amruthasangeeth Does not handling the not admin case in this pull request helps you in something? Any reason you have not handled this case with these changes?